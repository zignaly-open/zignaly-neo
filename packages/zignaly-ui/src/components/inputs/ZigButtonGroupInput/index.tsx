import React from "react";
import ZigButton, { ZigButtonGroup, ZigButtonProps } from "../ZigButton";
import ZigTypography from "../../display/ZigTypography";
import { Box } from "@mui/system";
import { ErrorMessage } from "../../display/ZigAlertMessage";

const ZigButtonGroupInput = ({
  value,
  label,
  error,
  options,
  onChange,
}: {
  value: string;
  options: {
    id?: string;
    value: string;
    label: string | JSX.Element;
    tooltip?: string;
    extraProps?: ZigButtonProps;
  }[];
  error?: string | JSX.Element;
  label?: string | JSX.Element;
  onChange: (value: string) => void;
}) => {
  return (
    <>
      {typeof label === "string" && (
        <Box sx={{ mb: 1.25 }}>
          <ZigTypography variant="body1" color="neutral200">
            {label}
          </ZigTypography>
        </Box>
      )}

      {!!label && typeof label !== "string" && label}

      <ZigButtonGroup>
        {options.map((o) => (
          <ZigButton
            id={o.id}
            onClick={() => onChange(o.value)}
            key={o.value}
            active={value === o.value}
            variant={"outlined"}
            {...(o.extraProps || {})}
          >
            {o.label}
          </ZigButton>
        ))}
      </ZigButtonGroup>

      {!!error && (
        <Box sx={{ marginTop: "3px" }}>
          <ErrorMessage text={error} />
        </Box>
      )}
    </>
  );
};

export default ZigButtonGroupInput;
