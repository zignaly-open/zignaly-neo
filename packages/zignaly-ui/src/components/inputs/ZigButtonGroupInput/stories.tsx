import React, { useMemo, useState } from "react";
import { Meta, Story } from "@storybook/react";
import ZigButtonGroupInput from "./index";

export default {
  title: "Inputs/ZigButtonGroupInput",
  component: ZigButtonGroupInput,
  argTypes: {},
} as Meta<typeof ZigButtonGroupInput>;

const Template: Story = () => {
  const [value, setValue] = useState("1");
  const options = useMemo(
    () => [
      { label: "Value #1", value: "1" },
      { label: "Value #2", value: "2" },
      { label: "Value #3", value: "3", extraProps: { tooltip: "Invalid value" } },
    ],
    [],
  );
  return (
    <>
      <ZigButtonGroupInput
        onChange={(v) => {
          setValue(v);
        }}
        value={value}
        label={"Select somehting"}
        error={+value === 3 ? "Can not be 3" : undefined}
        options={options}
      />
    </>
  );
};

export const Default = Template.bind({});
