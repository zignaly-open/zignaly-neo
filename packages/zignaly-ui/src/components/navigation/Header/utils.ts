// TODO: Move this for config or theme folder.
export const headerConfig = {
  height: "52px",
};
