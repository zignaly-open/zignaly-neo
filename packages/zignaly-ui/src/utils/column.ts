import styled from "styled-components";

export const Column = styled.div<{
  gap?: number;
  justifyContent?: "center" | "end" | "start" | "space-between";
  alignItems?: "center" | "end" | "start" | "space-between";
  textAlign?: "end" | "center" | "start";
  height?: string;
  flex?: number;
  width?: string;
}>`
  display: flex;
  flex-direction: column;
  flex: ${(props) => props.flex};
  gap: ${(props) => props.gap}px;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  text-align: ${(props) => `${props.textAlign}`};
  height: ${(props) => `${props.height}`};
  width: ${(props) => `${props.width}`};
`;
