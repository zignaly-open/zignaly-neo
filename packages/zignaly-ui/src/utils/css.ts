import { css } from "styled-components";

export const NiceScrollbar = css`
  &::-webkit-scrollbar {
    width: 6px;
  }

  /* Track */
  &::-webkit-scrollbar-track {
    background: rgba(255, 255, 255, 0.05);
    border-radius: 12px;
  }

  /* Handle */
  &::-webkit-scrollbar-thumb {
    background: rgba(255, 255, 255, 0.1);
    border-radius: 12px;
  }

  /* Handle on hover */
  &::-webkit-scrollbar-thumb:hover {
    background: rgba(255, 255, 255, 0.15);
  }
`;
