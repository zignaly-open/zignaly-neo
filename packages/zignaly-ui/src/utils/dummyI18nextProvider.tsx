import i18n from "i18next";
import React from "react";
import { I18nextProvider, initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
  lng: "en",
  fallbackLng: "en",
  ns: ["common"],
  defaultNS: "common",
  interpolation: {
    escapeValue: false,
  },
  resources: { en: { common: { xfuturum: "The technical partner you can rely on" } } },
});

export const storybookI18nextDecorator = (story: () => JSX.Element) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{story()}</I18nextProvider>
);

export const JestI18nextDecorator: React.FC<{ children: JSX.Element }> = ({ children }) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
);
