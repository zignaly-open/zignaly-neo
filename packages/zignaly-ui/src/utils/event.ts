export const changeEvent = (name: string | undefined, value: any) => {
  return {
    target: {
      value,
      name,
    },
    type: "change",
  };
};
