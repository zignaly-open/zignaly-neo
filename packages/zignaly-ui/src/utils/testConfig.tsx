import { render } from "@testing-library/react";
import { ThemeProvider as ThemeProviderMui } from "@mui/material";
import { ThemeProvider, ThemeProviderMui as ThemeInheritorMui } from "../index";
import React from "react";
import { getZignalyThemeExport } from "../theme";
import mediaQuery from "css-mediaquery";
import { JestI18nextDecorator } from "./dummyI18nextProvider";

export const renderWithProvidersUi = (ui: JSX.Element, renderOptions = {}) => {
  const { mui, legacyStyledComponentsDoNotUse } = getZignalyThemeExport();
  return render(ui, {
    wrapper: ({ children }: { children: JSX.Element }) => (
      <JestI18nextDecorator>
        <ThemeProvider theme={legacyStyledComponentsDoNotUse}>
          <ThemeInheritorMui theme={mui}>
            <ThemeProviderMui theme={mui}>{children}</ThemeProviderMui>
          </ThemeInheritorMui>
        </ThemeProvider>
      </JestI18nextDecorator>
    ),
    ...renderOptions,
  });
};

const createMatchMedia = (width: number) => {
  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width }),
      media: "",
      addListener: () => {},
      removeListener: () => {},
      onchange: () => {},
      addEventListener: () => {},
      removeEventListener: () => {},
      dispatchEvent: () => true,
    };
  };
};

export const resizeScreenSize = (width: number) => {
  window.matchMedia = createMatchMedia(width);
};
