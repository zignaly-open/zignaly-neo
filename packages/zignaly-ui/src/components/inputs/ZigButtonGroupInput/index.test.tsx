import React from "react";
import ZigButtonGroupInput from "./index";
import { renderWithProvidersUi } from "../../../utils/testConfig";
import { fireEvent, waitFor } from "@testing-library/react";

const testOptions = [
  {
    id: "test-button1",
    value: "test1",
    label: "button 1",
  },
  {
    id: "test-button2",
    value: "test2",
    label: "button 2",
  },
];
const fiveTestOptions = [
  {
    id: "test-button1",
    value: "test1",
    label: "button 1",
  },
  {
    id: "test-button2",
    value: "test2",
    label: "button 2",
  },
  {
    id: "test-button3",
    value: "test3",
    label: "button 3",
  },
  {
    id: "test-button4",
    value: "test4",
    label: "button 4",
  },
  {
    id: "test-button5",
    value: "test5",
    label: "button 5",
  },
];
let mainValue = "test1";

describe("components/inputs/ZigButtonGroupInput", () => {
  describe("ZigButtonGroup rendering", () => {
    it("renders with 2 options", async () => {
      const { container } = renderWithProvidersUi(
        <ZigButtonGroupInput value={mainValue} options={testOptions} onChange={() => {}} />,
      );

      const button1 = container.querySelector("#test-button1") as Element;
      const button2 = container.querySelector("#test-button2") as Element;
      expect(container).toMatchSnapshot();
      expect(button1).toBeVisible();
      expect(button2).toBeVisible();
    });

    it("renders with more than 2 options", async () => {
      const { container } = renderWithProvidersUi(
        <ZigButtonGroupInput value={mainValue} options={fiveTestOptions} onChange={() => {}} />,
      );

      const button1 = container.querySelector("#test-button1") as Element;
      const button2 = container.querySelector("#test-button2") as Element;
      const button3 = container.querySelector("#test-button3") as Element;
      const button4 = container.querySelector("#test-button4") as Element;
      const button5 = container.querySelector("#test-button5") as Element;
      expect(button1).toBeVisible();
      expect(button2).toBeVisible();
      expect(button3).toBeVisible();
      expect(button4).toBeVisible();
      expect(button5).toBeVisible();
    });
  });

  describe("ZigButtonGroup actions", () => {
    it("changes active value", async () => {
      const { container, rerender } = renderWithProvidersUi(
        <ZigButtonGroupInput
          value={mainValue}
          options={testOptions}
          onChange={(v) => {
            mainValue = v;
          }}
        />,
      );

      const button1 = container.querySelector("#test-button1") as Element;
      const button2 = container.querySelector("#test-button2") as Element;
      expect(button1).toHaveClass("MuiButton-active");
      expect(button2).not.toHaveClass("MuiButton-active");
      fireEvent.click(button2);
      await waitFor(() => {
        rerender(
          <ZigButtonGroupInput
            value={mainValue}
            options={testOptions}
            onChange={(v) => {
              mainValue = v;
            }}
          />,
        );
        const updatedButton1 = container.querySelector("#test-button1") as Element;
        const updatedButton2 = container.querySelector("#test-button2") as Element;
        expect(updatedButton2).toHaveClass("MuiButton-active");
        expect(updatedButton1).not.toHaveClass("MuiButton-active");
      });
    });
  });
});
