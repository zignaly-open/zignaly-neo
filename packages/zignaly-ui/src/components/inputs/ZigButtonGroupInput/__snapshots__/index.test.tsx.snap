// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/inputs/ZigButtonGroupInput ZigButtonGroup rendering renders with 2 options 1`] = `
<div>
  <div
    class="MuiButtonGroup-root MuiButtonGroup-outlined css-1oyhkbl-MuiButtonGroup-root"
    role="group"
  >
    <button
      class="MuiLoadingButton-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root undefined MuiButton-active MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary css-uqq2v3-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
      data-testid="test-button1"
      id="test-button1"
      tabindex="0"
      type="button"
    >
      button 1
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiLoadingButton-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary css-uqq2v3-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
      data-testid="test-button2"
      id="test-button2"
      tabindex="0"
      type="button"
    >
      button 2
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
